dts/arm/nxp/nxp_imx6sx_m4.dtsi

```diff
@@ -267,6 +267,17 @@
 			status = "disabled";
 		};
 
+		iomuxc: iomuxc@420e0000 {
```

Comment:
note: could this node be `pinctrl`?

Reply:
No, this node needs to remain as `iomuxc`. This pin control implementation is designed the same way as the implementation for the RT10xx/RT11xx, since the peripheral used for pin control is very similar.

The `iomuxc` node is referenced by the pin control definitions file in NXP's HAL (for the imx6sx, this file is `mimx6sx-pinctrl.dtsi`). The pin control definitions file defines all possible pinmux selections. So the `iomuxc` node will be populated with these pinmux options, and the "nxp,imx-iomuxc" compatible binding defines the format for these options.

The `pinctrl` node is populated with the pin control selections, defined per board. Unlike the pinmux options, which use the same format and DTS binding for all iMX SOCs, the available configuration fields for iMX SOCs differ between product families, necessitating several different DTS bindings. This difference here is the reason I chose to keep the `iomuxc` node generic, and populate it with the pinmux options, then populate the child `pinctrl` node with pinmux selections.